dataset/dataset.go

```diff
@@ -652,6 +659,57 @@ func (c *Client) GetInstances(ctx context.Context, userAuthToken, serviceAuthTok
 	return
 }
 
+func (c *Client) GetInstancesInBatches(ctx context.Context, userAuthToken, serviceAuthToken, collectionID string, vars url.Values, batchSize, maxWorkers int) (instances Instances, err error) {
+
+	// Function to aggregate items.
+	// For the first received batch, as we have the total count information, will initialise the final structure of items with a fixed size equal to TotalCount.
+	// This serves two purposes:
+	//   - We can guarantee, even with concurrent calls, that values are returned in the same order that the API defines, by offsetting the index.
+	//   - We do a single memory allocation for the final array, making the code more memory efficient.
+	var processBatch InstancesBatchProcessor = func(b Instances) (abort bool, err error) {
+		if len(instances.Items) == 0 { // first batch response being handled
+			instances.TotalCount = b.TotalCount
+			instances.Items = make([]Instance, b.TotalCount)
+			instances.Count = b.TotalCount
+		}
+		for i := 0; i < len(b.Items); i++ {
+			instances.Items[i+b.Offset] = b.Items[i]
+		}
+		return false, nil
+	}
+
+	// call dataset API GetInstances in batches and aggregate the responses
+	if err := c.GetInstancesBatchProcess(ctx, userAuthToken, serviceAuthToken, collectionID, vars, processBatch, batchSize, maxWorkers); err != nil {
+		return Instances{}, err
+	}
+
+	return instances, nil
+}
+
+// GetInstancesBatchProcess gets the instances from the dataset API in batches, calling the provided function for each batch.
+func (c *Client) GetInstancesBatchProcess(ctx context.Context, userAuthToken, serviceAuthToken, collectionID string, vars url.Values, processBatch InstancesBatchProcessor, batchSize, maxWorkers int) error {
+
+	// for each batch, obtain the dimensions starting at the provided offset, with a batch size limit,
+	// or the subste of IDs according to the provided offset, if a list of optionIDs was provided
```

Comment:
the second part of the comment can be removed (the subset of IDs is only for dimension options)